### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by gem-sethiamit**
Hi here is my demo(feature) branch.

### CODE REVIEW

File: src/main/java/com/example/DsStore/controller/CustomerController.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	@Autowired
+	private CustomerService customerService;
+
+	//POST - create customer
+	@PostMapping("/")
+	public ResponseEntity<Customer> createCustomer(@RequestBody Customer customer){
+		Customer createCustomer = this.customerService.createCustomer(customer);
+		return new ResponseEntity<>(createCustomer,HttpStatus.CREATED);
+	}
+
```
Comment (abhi-saini-gemini) [inline src/main/java/com/example/DsStore/controller/CustomerController.java:35]: Let's do a practice of autoformatting code before committing. 
IntelliJ: CTRL + ALT + F
Eclipse: CTRL + Shift + F

File: src/main/java/com/example/DsStore/controller/CustomerController.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	//DELETE - delete customer
+		@DeleteMapping("/{customerId}")
+		public ResponseEntity<ApiResponse> deleteUser(@PathVariable Integer customerId){
+			this.customerService.deleteCustomer(customerId);
+			return new ResponseEntity<ApiResponse>(new ApiResponse("Customer deleted successfully",true),HttpStatus.OK);
+		}
+
+		//GET ALL - get all customers 
+		@GetMapping("/")
+		public ResponseEntity<List<Customer>> getAllGoals(){
```
Comment (abhi-saini-gemini) [inline src/main/java/com/example/DsStore/controller/CustomerController.java:52]: `getAllGoals()` , `getSingleGoal()`. Method name seems to not match with domain. Let's rename them based on our entity.

File: src/main/java/com/example/DsStore/entities/Customer.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.example.DsStore.entities;
+
+
+import jakarta.persistence.*;
```
Comment (abhi-saini-gemini) [inline src/main/java/com/example/DsStore/entities/Customer.java:4]: IDEs have a default behaviour of replacing multiple imports of same library by star (*), also called as wildcards.
Could you search online and see what are people's opinion on this?

Let me know here.

**Review by abhi-saini-gemini — CHANGES_REQUESTED** [review]
Hi Amit, could we take a look at the following pointers and check back?
1. Code formatting.
2. method name in controller.
3. Wildcard import (*)

**Comment by abhi-saini-gemini** [comment]
> Hi here is my demo(feature) branch.

Some things to keep in mind when creating merge request/pull request.

https://opensource.com/article/18/6/anatomy-perfect-pull-request

File: src/main/java/com/example/DsStore/controller/CustomerController.java (hunk 1 above)
Comment (gem-sethiamit) [reply src/main/java/com/example/DsStore/controller/CustomerController.java:35]: sure

File: src/main/java/com/example/DsStore/controller/CustomerController.java (hunk 2 above)
Comment (gem-sethiamit) [reply src/main/java/com/example/DsStore/controller/CustomerController.java:52]: right

File: src/main/java/com/example/DsStore/entities/Customer.java (hunk 3 above)
Comment (gem-sethiamit) [reply src/main/java/com/example/DsStore/entities/Customer.java:4]: There are both advantages and disadvantages of wildcards:
adv:- clean code
dis adv:- name conflicts, *  import all the classes under that package most of which may not be even.
Hence most people **view wildcards as bad coding practice.**

**Pull request merged**